### PR TITLE
Add exit option in withdraw dialog

### PR DESCRIPTION
**Pull request by oskarvu**
Closes FRO-983

### CODE REVIEW

**Comment by EmilMalanczak** [comment]
are we able to adjust form input value in the stories? This switch should be displayed only when MAX is selected
image

**Comment by oskarvu** [comment]
@EmilMalanczak We normally don't play around with form at stories. It would complicate them a lot, and still won't give you a perfect behavior. We're displaying view based on props passed to it. Switch is checked based on prop view is getting, and it's reflected in story arguments. I think that's good enough.

**Review by EmilMalanczak — APPROVED** [review]

**Comment by yivlad** [comment]
@oskarvu I think modifying input values makes to reflect the state makes sense, we can just pass `defaultValues` to form in stories
```ts
const form = useForm({
    defaultValues: {
        symbol: args.assetsFields.selectedAsset.token.symbol,
        value: args.assetsFields.selectedAsset.value,
        isMaxSelected: args.assetsFields.selectedAsset.value ===  args.assetsFields.maxValue?.toFixed()
      }
    }) as any
```

File: packages/app/src/features/farm-details/dialogs/unstake/components/transaction-overview/TransactionOutcome.tsx [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
+  const outcomeTokenAmount = outcomeToken.format(outcomeTokenRouteItem.value, { style: 'auto' })
+  const outcomeTokenUsdValue = outcomeToken.formatUSD(outcomeTokenRouteItem.usdValue)
+  const earnedRewardsAmount = rewardToken.format(earnedRewards, { style: 'auto' })
+  const earnedRewardsUsdValue = rewardToken.formatUSD(earnedRewards)
+
+  const unstakeText = `${outcomeTokenAmount} ${outcomeToken.symbol} worth ${outcomeTokenUsdValue}`
+  const unstakeTextMobile = `${outcomeTokenAmount} ${outcomeToken.symbol}`
+  const exitText = `${outcomeTokenAmount} ${outcomeToken.symbol} (${outcomeTokenUsdValue}) + ${earnedRewardsAmount} ${rewardToken.symbol} (${earnedRewardsUsdValue})`
+  const exitTextMobile = `${outcomeTokenAmount} ${outcomeToken.symbol} + ${earnedRewardsAmount} ${rewardToken.symbol}`
+
```
Comment (yivlad) [inline packages/app/src/features/farm-details/dialogs/unstake/components/transaction-overview/TransactionOutcome.tsx:29]: I'd use the pattern with self invoked function
```ts
const [textContent, textContentMobile] = (() => {
  if (isExiting) {
    const exitText = `${outcomeTokenAmount} ${outcomeToken.symbol} (${outcomeTokenUsdValue}) + ${earnedRewardsAmount} ${rewardToken.symbol} (${earnedRewardsUsdValue})`
    const exitTextMobile = `${outcomeTokenAmount} ${outcomeToken.symbol} + ${earnedRewardsAmount} ${rewardToken.symbol}`
    return [exitText, exitTextMobile]
  }
  const unstakeText = `${outcomeTokenAmount} ${outcomeToken.symbol} worth ${outcomeTokenUsdValue}`
  const unstakeTextMobile = `${outcomeTokenAmount} ${outcomeToken.symbol}`
  return [unstakeText, unstakeTextMobile]
})()
```

**Comment by oskarvu** [comment]
@yivlad About form default values. Let's do this, but at the same time we should make it a convention from now on (like passing empty form was convention before).

File: packages/app/src/features/farm-details/dialogs/unstake/logic/useUnstakeDialog.ts [hunk 2]
```diff
@@ -96,6 +107,15 @@ export function useUnstakeDialog({ farm, initialToken }: UseStakeDialogParams):
       actionsEnabled,
       goToSuccessScreen: () => setPageStatus('success'),
     },
+    exitFarmSwitchInfo: {
+      showSwitch: formValues.isMaxSelected,
+      onSwitch: () => setExitFarmSwitchChecked((exitFarmSwitchChecked) => !exitFarmSwitchChecked),
+      checked: exitFarmSwitchChecked,
+      reward: {
+        token: farm.rewardToken,
+        value: farm.earned,
```
Comment (yivlad) [inline packages/app/src/features/farm-details/dialogs/unstake/logic/useUnstakeDialog.ts:116]: Replace with value returned from `calculateReward` function

Comment (yivlad) [reply packages/app/src/features/farm-details/dialogs/unstake/logic/useUnstakeDialog.ts:116]: Same for SuccessView

File: packages/app/src/features/farm-details/dialogs/unstake/views/UnstakeView.stories.tsx [hunk 3]
```diff
@@ -99,10 +106,10 @@ const meta: Meta<typeof UnstakeView> = {
     assetsFields: {
       selectedAsset: {
         token: tokens.USDS,
-        balance: NormalizedUnitNumber(50000),
+        balance: NormalizedUnitNumber(50_000),
         value: '2000',
       },
-      maxValue: NormalizedUnitNumber(50000),
+      maxValue: NormalizedUnitNumber(50_000),
```
Comment (yivlad) [inline packages/app/src/features/farm-details/dialogs/unstake/views/UnstakeView.stories.tsx:112]:
```suggestion
      maxValue: NormalizedUnitNumber(50_000),
      maxSelectedFieldsName: 'isMaxSelected'
```

Comment (yivlad) [reply packages/app/src/features/farm-details/dialogs/unstake/views/UnstakeView.stories.tsx:112]: Also the value should be equal to maxValue if we want MAX button to be pressed

**Review by yivlad — APPROVED** [review]

**Pull request merged**